### PR TITLE
article content type

### PR DESCRIPTION
**Pull request by katlazarova**

### CODE REVIEW

File: config/sync/core.entity_form_display.node.article.default.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
+uuid: 94f4f567-c039-46c9-91f9-1e7b8e34d508
+langcode: en
+status: true
+dependencies:
+  config:
+    - field.field.node.article.field_a_body
+    - field.field.node.article.field_a_content_author
+    - field.field.node.article.field_a_headline
```
Comment (ElXyfer) [inline config/sync/core.entity_form_display.node.article.default.yml]: Delete this field you can reuse the title field and rename it to headline in the content type settings

File: config/sync/field.field.node.article.field_a_content_author.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+entity_type: node
+bundle: article
+label: 'Content Author'
+description: ''
+required: false
+translatable: false
+default_value: {  }
+default_value_callback: ''
+settings: {  }
+field_type: string
```
Comment (ElXyfer) [inline config/sync/field.field.node.article.field_a_content_author.yml]: Change this to be a select list like recipe content type and also reuse the `content_author_allowed_values_function`

File: config/sync/core.entity_form_display.node.article.default.yml (hunk 1 above)
Comment (katlazarova) [reply config/sync/core.entity_form_display.node.article.default.yml]: done

File: config/sync/field.field.node.article.field_a_content_author.yml (hunk 2 above)
Comment (katlazarova) [reply config/sync/field.field.node.article.field_a_content_author.yml]: done

File: web/modules/custom/pineapple_recipe/pineapple_general.module [hunk 3]
```diff
@@ -29,13 +29,19 @@ function pineapple_recipe_help($route_name, RouteMatchInterface $route_match) {
 /**
  * Implements hook_form_alter().
  */
-function pineapple_recipe_form_alter(&$form, \Drupal\Core\Form\FormStateInterface $form_state, $form_id) {
+function pineapple_form_alter(&$form, \Drupal\Core\Form\FormStateInterface $form_state, $form_id) {
```
Comment (ElXyfer) [inline web/modules/custom/pineapple_recipe/pineapple_general.module]: This should be `pineapple_general_form_alter`

File: config/sync/core.entity_view_display.node.article.default.yml [hunk 4]
```diff
@@ -0,0 +1,67 @@
+uuid: dd960c84-ac23-488b-bb5c-1b9031573a45
+langcode: en
+status: true
+dependencies:
+  config:
+    - field.field.node.article.field_a_body
+    - field.field.node.article.field_a_content_author
+    - field.field.node.article.field_a_hero_image
+    - field.field.node.article.field_a_summary
+    - field.field.node.article.field_a_teaser_image
```
Comment (ElXyfer) [inline config/sync/core.entity_view_display.node.article.default.yml]: These fields are in the wrong order in the form display, they should be in the correct order as per task list

File: web/modules/custom/pineapple_recipe/pineapple_general.module (hunk 3 above)
Comment (ElXyfer) [reply web/modules/custom/pineapple_recipe/pineapple_general.module]: core.extension file, pineapple_recipe file and their contents should be updated to have the module renamed to pineapple_general

Comment (ElXyfer) [reply web/modules/custom/pineapple_recipe/pineapple_general.module]: And references to recipe in this file should be changed to general

Comment (katlazarova) [reply web/modules/custom/pineapple_recipe/pineapple_general.module]: done

File: config/sync/core.entity_view_display.node.article.default.yml (hunk 4 above)
Comment (katlazarova) [reply config/sync/core.entity_view_display.node.article.default.yml]: done

**Pull request merged**